src/execution/IncrementalPublisher.ts

```diff
@@ -78,7 +74,10 @@ export interface FormattedInitialIncrementalExecutionResult<
 export interface SubsequentIncrementalExecutionResult<
   TData = unknown,
   TExtensions = ObjMap<unknown>,
-> extends Partial<IncrementalUpdate<TData, TExtensions>> {
+> {
+  pending?: ReadonlyArray<PendingResult>;
+  incremental?: ReadonlyArray<IncrementalResult<TData, TExtensions>>;
+  completed?: ReadonlyArray<CompletedResult>;
```

Comment:
I think we should keep these required args? better to be explicit imo.


```suggestion
  pending: ReadonlyArray<PendingResult>;
  incremental: ReadonlyArray<IncrementalResult<TData, TExtensions>>;
  completed: ReadonlyArray<CompletedResult>;
```

Reply:
This is a bit tricky to follow, but I think it's correct. `SubsequentIncrementalExecutionResult` is part of the new return type of `execute`, and so these keys are optional => if there isn't anything newly pending, we do not send a pending key at all rather than sending a key with an empty array.

Previously, we were using within our `IncrementalPublisher` class an intermediate result type we called `IncrementalUpdate` to collect these entries, and then only sending them if they were populated.

So we had `SubsequentIncrementalExecutionResult` extend `Partial<IncrementalUpdate>`. That itself was confusing, and so we got rid of `IncrementalUpdate` entirely and just put `_incremental` and `_completed` on the `IncrementalPublisher` class instance itself.